lib/rules/name-property-casing.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * @fileoverview Name property casing for consistency purposes
+ * @author Armano
+ */
+'use strict'
+
+const utils = require('../utils')
+
+function kebabCase (str) {
+  return str.replace(/([a-z])([A-Z])/g, match => match[0] + '-' + match[1]).replace(/\s+/g, '-').toLowerCase()
+}
+
+function camelCase (str) {
+  return str.replace(/(?:^\w|[A-Z]|\b\w)/g, (letter, index) => index === 0 ? letter.toLowerCase() : letter.toUpperCase()).replace(/[\s-]+/g, '')
+}
+
+function pascalCase (str) {
+  str = camelCase(str)
+  return str.length > 0 ? str.charAt(0).toUpperCase() + str.slice(1) : ''
+}
+
+function convertCase (str, caseType) {
+  if (caseType === 'kebab-case') {
```

Comment:
Ideal case for `switch/case`. Or better you can create map with converters:

```
const convertersMap = {
  'kebab-case': kebabCase,
  'camelCase': camelCase,
  'PascalCase': pascalCase
}

function getConverter(name) {
  return convertersMap[name] || pascalCase
}
```

And use it like so:
```
function convertCase (str, caseType) {
  return getConverter(caseType)(str)
}
```

WDYT?